### PR TITLE
corro-agent: add API endpoint to query row count for set of tables

### PR DESCRIPTION
**Pull request by spacekookie**

### CODE REVIEW

**Comment by jeromegn** [comment]
"Sync" here is a little confusing to me. I'd expect to get something related to the sync protocol like that last nodes synced with and how many changes were synced, etc.

Maybe something related to "metrics" or "stats" or "reports" instead? I'm not sure.

**Comment by spacekookie** [comment]
Hm, do we expose explicit sync events via the API yet? But I do agree, it might be a bit confusing. We could go with `table_status` for this particular endpoint, which would be very explicit in what it does.

I went for a bit more of a generic name in the hopes to add more functionality to it down the line. But we can also cross future naming bridges when we get to them :sweat_smile:

File: crates/corro-agent/src/api/public/mod.rs [hunk 1]
```diff
@@ -656,6 +661,58 @@ pub async fn api_v1_db_schema(
...
 
+/// Query the sync status of the current node
+///
+/// Currently this endpoint only supports querying the row count for a
+/// selection of provided tables.  In the future we may provide more
+/// sophisticated methods to determine whether corrosion is "caught
+/// up" or not.
+pub async fn api_v1_sync_status(
+    Extension(agent): Extension<Agent>,
+    axum::extract::Json(tables): axum::extract::Json<Vec<String>>,
```
[the remark this answers is not in the document]
Comment (spacekookie) [reply crates/corro-agent/src/api/public/mod.rs:672]: Ah, woops. No reason, just the function I copied the line from was a mutation

File: crates/corro-agent/src/api/public/mod.rs [hunk 2]
```diff
@@ -656,6 +661,58 @@ pub async fn api_v1_db_schema(
...
+) -> (StatusCode, axum::Json<StatusResponse>) {
+    async fn count_table_lengths(agent: &Agent, tables: Vec<String>) -> eyre::Result<i64> {
+        debug!("Querying row count for {} tables", tables.len());
+        let conn = agent.pool().write_normal().await.unwrap();
```
[the remark this answers is not in the document]
Comment (spacekookie) [reply crates/corro-agent/src/api/public/mod.rs:676]: I wasn't super sure about this. I was thinking that we would probably just add more fields to the return type. But maybe making the input properly typed is a good idea nonetheless

**Review by pborzenkov — APPROVED** [review]

**Comment by spacekookie** [comment]
(just removed the unused `e`)

**Pull request merged**